clients/src/main/java/org/apache/kafka/common/record/RecordsUtil.java

```diff
@@ -80,9 +81,11 @@
         ByteBuffer buffer = ByteBuffer.allocate(totalSizeEstimate);
         long temporaryMemoryBytes = 0;
         int numRecordsConverted = 0;
+
         for (RecordBatchAndRecords recordBatchAndRecords : recordBatchAndRecordsList) {
             temporaryMemoryBytes += recordBatchAndRecords.batch.sizeInBytes();
             if (recordBatchAndRecords.batch.magic() <= toMagic) {
+                buffer = Utils.ensureCapacity(buffer, buffer.limit() + recordBatchAndRecords.batch.sizeInBytes());
```

Comment:
Should this be `buffer.position()` instead of `buffer.limit()`?

Reply:
doh, yes. thanks!